### PR TITLE
Fix: celo-specific transaction (CIP64) regressions

### PR DESCRIPTION
**Pull request by nicolasbrugneaux**
This PR aims to fix the issues reported by @0xarthurxyz in https://github.com/celo-org/txtypes/issues/1

This refactors the different checks for CIP42 and CIP64 by using more robust mechanism than truthy/falsy values.
It also removes the tests that added type: 'cipxyz' unnecessarily and by doing so forcing the tx to be correct even though the transaction-type inference not detect it as such. (eg: a cip42 without gatewayFee).

It also refactors some of the tests that checked for a hardcoded raw serialized string (keeps it in a couple place for sanity) but uses the parser to self-check itself.

[Regressions Introduced via #1379]

[Closes celo-org/txtypes/issues/1]

[Original PR clabs-co/viem/pull/6]

---

## PR-Codex overview
This PR focuses on fixing an issue where Celo CIP64 transactions were being recognized as EIP1559 transactions.

### Detailed summary
- Fix issue with Celo CIP64 transactions being recognized as EIP1559 transactions
- Import `isCIP42` and `isCIP64` functions in `formatters.ts`
- Update `formattersCelo` object in `formatters.ts` to use `isCIP42` and `isCIP64` functions
- Update `CeloTransactionSerializableBase` type in `types.ts`
- Import `isCIP42`

> The following files were skipped due to too many changes: `src/chains/celo/serializers.ts`, `src/chains/celo/utils.test.ts`, `src/chains/celo/serializers.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/chains/celo/utils.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+//////////////////////////////////////////////////////////////////////////////
+// Utilities
+
```
Comment (jxom) [inline src/chains/celo/utils.ts:3]:
```suggestion
```

File: src/chains/celo/utils.ts
Comment (jxom) [inline src/chains/celo/utils.ts:1]: Can we add test coverage to this file?

**Comment by nicolasbrugneaux** [comment]
Totally, good call. Will add tests for it in the morning EU.

File: src/chains/celo/utils.ts
Comment (nicolasbrugneaux) [reply src/chains/celo/utils.ts:1]: Done :)

**Pull request merged**